client/components/theme/index.jsx

```diff
@@ -344,3 +350,8 @@ export default connect(
 	},
 	{ recordTracksEvent, setThemesBookmark, updateThemes }
 )( localize( Theme ) );
+
+export default ( props ) => {
+	const { shouldLimitGlobalStyles } = usePremiumGlobalStyles( props.siteId );
+	return <ConnectedTheme { ...props } shouldLimitGlobalStyles={ shouldLimitGlobalStyles } />;
+};
```

Comment:
Is there a reason why we're not including the `shouldLimitGlobalStyle` prop in the `connect` call?

Wouldn't the following DIFF also allow us to add the `shouldLimitGlobalStyle` prop without introducing an additional level of indirection? Or am I missing something?

```DIFF
diff --git a/client/components/theme/index.jsx b/client/components/theme/index.jsx
index 8a94fb6cab..7fc2f0e850 100644
--- a/client/components/theme/index.jsx
+++ b/client/components/theme/index.jsx
@@ -332,12 +332,13 @@ export class Theme extends Component {
        }
 }
 
-const ConnectedTheme = connect(
+export default connect(
        ( state, { theme, siteId } ) => {
                const {
                        themes: { themesUpdate },
                } = state;
                const { themesUpdateFailed, themesUpdating, themesUpdated } = themesUpdate;
+               const { shouldLimitGlobalStyles } = usePremiumGlobalStyles( siteId );
                const isExternallyManagedTheme = getIsExternallyManagedTheme( state, theme.id );
 
                return {
@@ -346,12 +347,8 @@ const ConnectedTheme = connect(
                        isUpdated: themesUpdated && themesUpdated.indexOf( theme.id ) > -1,
                        isExternallyManagedTheme,
                        siteSlug: getSiteSlug( state, siteId ),
+                       shouldLimitGlobalStyles,
                };
        },
        { recordTracksEvent, setThemesBookmark, updateThemes }
 )( localize( Theme ) );
-
-export default ( props ) => {
-       const { shouldLimitGlobalStyles } = usePremiumGlobalStyles( props.siteId );
-       return <ConnectedTheme { ...props } shouldLimitGlobalStyles={ shouldLimitGlobalStyles } />;
-};
(END)

```

Reply:
Reacts hooks can be used only on functional classes. You actually had to do something similar in https://github.com/Automattic/wp-calypso/pull/69242/files#diff-bc79a829db32a21a70056887cf8a6bd4ebce6890c1a8fd3091258dba0610b680R809 😄

Reply:
Then we could add the hook to the actual component, right?

Reply:
Nope, because it's a class component, not a funcional component.